components/Markdown.tsx

```diff
@@ -4,22 +4,26 @@ import remarkGfm from "remark-gfm";
 import remarkRehype from "remark-rehype";
 import rehypeStringify from "rehype-stringify";
 import clsx from "clsx";
-export default async function Markdown(props: {
+import { useMemo } from "react";
+
+export default function Markdown(props: {
   children: string;
   className?: string;
 }) {
-  const result = await unified()
-    .use(remarkParse)
-    .use(remarkGfm)
-    .use(remarkRehype)
-    .use(rehypeStringify)
-    .process(props.children || "");
+  const processedMarkdown = useMemo(() => {
+    return unified()
+      .use(remarkParse)
+      .use(remarkGfm)
+      .use(remarkRehype)
+      .use(rehypeStringify)
+      .processSync(props.children || ""); // Using processSync for efficiency
+  }, [props.children]);
```

Comment:
Why is this useMemo added? 
We only have `className` and `children` that could cause the component to re-render. useMemo does not have any effect here right? (Correct me if I'm wrong)

Reply:
I am finding out why the page is rendered after each filter.
So I find that there is an issue with markdown.tsx 
and in that there are two things we are using one is `unified()` and another is `dangerouslySetInnerHTML` so that is lead to re-render (I got to know by bard and ChatGPT) and they give the solution of use memo .

Reply:
`useMemo` is not what solved the issue here. Making the component a sync component instead of an async component is what solved the issue. Making it sync without useMemo alone solves the issue.